packages/ui/src/forum/hooks/useForumMultiQueryCategoryBreadCrumbs.ts

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useMemo, useState } from 'react'
+
+import { useGetForumCategoryBreadcrumbLazyQuery } from '../queries'
+import { asSubCategory, CategoryBreadcrumb } from '../types'
+
+// The goal of this hook is to utilize apollo cache to be faster than `useForumCategoryBreadcrumbs`
+// when there are a lot of breadcrumbs to show, like on search results
```

Comment:
The previous `useForumCategoryBreadcrumbs` is much better for fetching a single breadcrumbs because it does it in only one query. But because there can many results to render with different breadcrumbs but which are probably sharing some root categories, I thought that for this particular case, it could be better to have slightly more but shorter cacheable queries.

Eg:
There are several results from `C1 > C2 > thread1`, from `C1 > C2 > C3 > thread2`, and from `C1 > thread3`
For their breadcrumbs:
-  `C1 > C2 > thread1`: queries `C2`, and gets `C1` from the cache
- `C1 > C2 > C3 > thread2`: queries `C3`, and gets `C2` and `C1` from the cache
- `C1 > thread3`: queries `C1`

WDYT ?

Reply:
I think that the idea is pretty neat – thanks for the explanation. If we could discuss that earlier I would opt-in for using the existing hook and adding optimization if or when we'd saw a need for that. Just because it is hard to tell how real-world data will present. For instance, if most of the results are from a single category, then they should be cached already. OTOH it might be a case that most of the results came from single parents, but multiple sub-categories. Then, our solution will be better.

But since you already implemented this let's keep that and see what will happen.

ps.: Another solution could be to add breadcrumbs to the forum thread/category in Hydra itself as it is used in multiple places.